alacritty/src/renderer/rects.rs

```diff
@@ -190,3 +195,216 @@ impl RenderLines {
         }
     }
 }
+
+/// Shader sources for rect rendering program.
+pub static RECT_SHADER_F_PATH: &str = concat!(env!("CARGO_MANIFEST_DIR"), "/res/rect.f.glsl");
+pub static RECT_SHADER_V_PATH: &str = concat!(env!("CARGO_MANIFEST_DIR"), "/res/rect.v.glsl");
+static RECT_SHADER_F: &str = include_str!("../../res/rect.f.glsl");
+static RECT_SHADER_V: &str = include_str!("../../res/rect.v.glsl");
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+struct Rgba {
+    r: u8,
+    g: u8,
+    b: u8,
+    a: u8,
+}
+
+/// Struct that stores vertex 2D coordinates and color for rect rendering.
+#[repr(C)]
+#[derive(Debug, Clone, Copy)]
+struct Vertex {
+    // Normalized screen coordinates.
+    // TODO these can certainly be i16.
```

Comment:
I think we've discussed and agreed that it can't really be i16 due to subpixel precision, etc? I'm not sure if we should care about it for plain rectangles, but whatever.

Reply:
I'd say making it i16 is possible, as all this subpixel stuff should not affect anything that much here. But it would also be helpful to also revise non-rendering portion of it also. Let's do this separately.